pkg/operator/controller/extension/admission/admission.go

```diff
@@ -208,11 +212,19 @@ func (d *deployment) createOrUpdateAdmissionVirtualClusterResources(ctx context.
 		}
 	}
 
-	renderedChart, err := virtualClusterClientSet.ChartRenderer().RenderArchive(archive, extension.Name, v1beta1constants.GardenNamespace, utils.MergeMaps(helmValues, gardenerValues))
+	namespace := virtualNamespace(extension)
```

Comment:
Shouldn't we also this namespace to create the `ServiceAccount` used for the access secret?

Reply:
This only works if we introduce another option for the `AccessSecret`. At the moment you can set the service account namespace by setting its `Class` to `shoot` ([ref](https://github.com/gardener/gardener/blob/fb4110f8c2570c9164fe54a835f3614e05ff4a20/pkg/utils/gardener/shoot.go#L392-L394)) only.
Otherwise, there is no other option to set the `serviceaccount.resources.gardener.cloud/namespace` annotation.

We could easily introduce an option like `WithServiceAccountNamespaceOverride`. However, that's not all. GRM watches `garden`, `kube-system` and `gardener-system-shoot-issuer` namespaces ([ref](https://github.com/gardener/gardener/blob/fb4110f8c2570c9164fe54a835f3614e05ff4a20/pkg/operator/controller/garden/garden/components.go#L399)) only. Thus, we would have to redeploy it every time an extension changes. This looks too complicated in my opinion.